internal/services/groups/group_resource.go

```diff
@@ -1065,6 +1100,29 @@ func groupResourceUpdate(ctx context.Context, d *schema.ResourceData, meta inter
 		}
 	}
 
+	if d.HasChange("administrative_unit_id") {
+		administrativeUnitClient := meta.(*clients.Client).AdministrativeUnits.AdministrativeUnitsClient
+		o, n := d.GetChange("administrative_unit_id")
+		oldAdministrativeUnitId := o.(string)
+		newAdministrativeUnitId := n.(string)
+		if newAdministrativeUnitId != "" {
+			members := msgraph.Members{
+				group.DirectoryObject,
+			}
+			members[0].ODataId = (*odata.Id)(utils.String(fmt.Sprintf("%s/v1.0/%s/directoryObjects/%s",
+				client.BaseClient.Endpoint, client.BaseClient.TenantId, d.Id())))
+			if _, err := administrativeUnitClient.AddMembers(ctx, newAdministrativeUnitId, &members); err != nil {
+				return tf.ErrorDiagF(err, "Could not add group to administrative unit with object ID: %q", newAdministrativeUnitId)
+			}
+		}
+		if oldAdministrativeUnitId != "" {
+			memberIds := []string{d.Id()}
+			if _, err := administrativeUnitClient.RemoveMembers(ctx, oldAdministrativeUnitId, &memberIds); err != nil {
+				return tf.ErrorDiagF(err, "Could not remove group from administrative unit with object ID: %q", oldAdministrativeUnitId)
+			}
+		}
```

Comment:
Whilst leaning on the diff is a useful shortcut, in this case we should look up the AU memberships for the group and remove all but the specified one (e.g. https://learn.microsoft.com/en-us/graph/api/group-list-memberof?view=graph-rest-1.0). This ensures that Terraform is declarative here in all cases, e.g. when importing groups, changing from multiple AUs to a single AU etc.

Reply:
I'm not sure what would be a good way to proceed here.
With the current state of the PR, it is possible to create a group within an AU1 and then use the `administrative_unit_member` resource to add the same group to AU2.

If we follow your suggestion and make Terraform declarative in this case, reapplying the group would remove the administrative unit membership of AU2. That would mean that it's not possible to both create a group within an AU and also add it to a second AU, which seems limiting to me. When creating this PR I was wondering on whether `administrative_unit_id` should instead be a slice of IDs, as that is closer to how it works in Azure AD. We could follow a similar approach to how `owners` work: create the group within one AU and add other AU memberships afterwards.

Reply:
Ah I see what you mean. In this case I think your suggestion would work best - if we change `adminsitrative_unit_id` to `administrative_unit_ids` this would alleviate the challenge for users of introducing conflicts with `azuread_administrative_unit_member`, and should still offer decent functionality. We'll note in the docs that permission will be needed to add/create in each of the specified AUs else group creation will fail.

Reply:
I have changed it to take multiple `administrative_unit_ids` now. Could you look it over again please?
Also I added a note in the permissions section.